api-registration/server/server.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import 'dotenv/config';
+import pg from 'pg';
+import argon2 from 'argon2';
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+app.use(express.json());
+
+async function checkUsername(username, res) {
+  const sql = `
+      SELECT "username" FROM "users"
+      WHERE "username" = $1;`;
+  const params = [username];
+  const result = await db.query(sql, params);
+  const user = result.rows[0];
+  if (user) {
+    throw new ClientError(400, `Username '${username}' already exists. Please try another username.`);
+  }
+}
+
+app.post('/api/auth/sign-up', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password) {
+      throw new ClientError(400, 'username and password are required fields');
+    }
+    await checkUsername(username, res);
```

Comment:
Nice error checking. But... Insert will fail if the username already exists, so you could just issue one query (the insert). When you host in a Cloud system, you frequently have to pay per query, so it's good to keep them to a minimum.